src/index.tsx

```diff
@@ -1,16 +1,118 @@
-import {createRoot} from 'react-dom/client';
-import {ThemeProvider} from '@gravity-ui/uikit';
+import {useCallback, useState, useEffect}  from 'react';
+import {Container, Row, Col} from '@gravity-ui/uikit';
 
-import {App} from './app';
+import {CallOut} from './callout';
+import {InputArea} from './input-area';
+import {OutputArea} from './output-area';
 
-import './index.css';
-import '@doc-tools/transform/dist/js/yfm.js';
+import {useTabs} from './useTabs';
+import {generateMD, generateHTML, generateTokens} from './generators';
+import persistRestore from './persistRestore';
 
-const container = document.getElementById('app');
-const root = createRoot(container as HTMLElement);
+import './styles.css';
 
-root.render(
-  <ThemeProvider theme="light"> 
-    <App />
-  </ThemeProvider>
-);
+(window as any).MonacoEnvironment = {
+    getWorker: (_, label: string) => {
+        if (label === 'json') {
+            return new Worker(
+                new URL('monaco-editor/esm/vs/language/json/json.worker?worker', 'monaco-worker'),
+            );
+        }
+        return new Worker(new URL('monaco-editor/esm/vs/editor/editor.worker?worker', 'monaco-worker'));
+    },
+};
+
+const App = () => {
+    return(<>
+        <Container>
+            <CallOut />
+            <Playground persistRestore={true} />
+        </Container>
+    </>)
+};
+
+export type PlaygroundProperties = {
+    content?: string;
+    tabs?: { id: string, title: string }[];
```

Comment:
we are allowing to choose from predefined tabs(with transformations that goes with them)

each id has transformations tied up to them: md2html(preview), md2html(raw), md2md, etc...

change type to

```
tabs?: Tabs[]

export type Tabs = {
  id: 'preview' | 'html' | 'markdown' | 'tokens';
  title: string;
}
```